plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -571,6 +573,23 @@ open class DefaultPageCreator(
         }
     }
 
+    /**
+     * Divergent elements, such as extensions for the same receiver, can have identical signatures
+     * if they are declared in different places. If such elements are shown on the same page together,
+     * they need to be rendered deterministically to have reproducible builds.
+     *
+     * For example, you can have three identical extensions, if they are declared as:
+     * 1) top-level in package A
+     * 2) top-level in package B
+     * 3) inside a companion object in package A/B
+     *
+     * @see divergentBlock
+     */
+    private fun sortDivergentElementsDeterministically(elements: List<Documentable>): List<Documentable> =
+        elements.takeIf { it.size > 1 }
```

Comment:
This is redundant. The same check is inside `sortedWith`.

Reply:
Are you sure? As far as I can see, `sortedWith` returns a brand new list for single-element lists

Reply:
It costs almost nothing.
It is consistent to have a new one.
e.g. 
```kt
val list = mutableListOf(...)
val doWeHaveANewListOrNot = sortDivergentElementsDeterministically(list)
list == doWeHaveANewListOrNot
// modify list
```

Reply:
Unless it's critical for you, I'd like to leave it as is, even if it costs almost nothing. This will be called for every type (annotation/class/interface/object/typealias), function and property, the overwhelming majority of which will be 1 element lists that don't need sorting. The check is simple enough to understand and shouldn't hurt.

Reply:
>  which will be 1 element lists that don't need sorting.

They will not be sorted. I have not understood what the problem with short-liver single-element lists is.

> The check is simple enough to understand and shouldn't hurt. 

I gave the example above when it can hurt.

Reply:
Originally, I also thought that this shortcut was redundant, but given the no-copy optimization consideration in case of single-element lists and the fact that this function is private and used only in one place where we know for sure that its input is not mutable, I don't think it can cause confusion. 

And to avoid it further, the function can be provided a kdoc describing its contract in case it is needed elsewhere, e.g. "returns the original list if it has one or zero elements"

Reply:
Given how this is a client app that often runs on free-tier CI nodes with limited resources, I believe these things can add up to something noticeable over time. Even if not, I don't think it hurts to the point where it needs to be deleted now. If it gets in the way after some refactoring - sure, I don't mind, we can remove it.

With or without the "optimization", I'd still like to have a separate function to have documentation, so we might as well check for size while we're at it.

Added documentation as proposed

Reply:
By the way, speaking about micro-optimization, you can use `MutableList<Documentable>` and in-place `sortWith`. It will allow to get more performance and avoid this heuristic.
It needs to replace `.groupBy{ it.name }` with `.groupByTo(mutableMapOf()) { it.name }`